### PR TITLE
DAOS-3553 bio: SCM RDMA update over DRAM buffer

### PR DESCRIPTION
**Pull request by NiuYawei**
Implement SCM RDMA update over DRAM buffer since the AEP RDMA update
performance isn't good at this moment. This feature can be enabled
by setting env BIO_BUFFERED_SCM_RDMA=1 on server.

Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### CODE REVIEW

File: src/bio/bio_buffer.c [hunk 1]
```diff
@@ -617,7 +647,48 @@ dma_map_one(struct bio_desc *biod, struct bio_iov *biov,
...
+void
+bio_memcpy(struct bio_desc *biod, uint16_t media, void *media_addr,
+	   void *addr, ssize_t n)
+{
+	struct umem_instance *umem = biod->bd_ctxt->bic_umem;
+
+	if (biod->bd_update && media == DAOS_MEDIA_SCM) {
+		/* NB: pmemobj_tx_commit will drain HW buffer */
+		pmemobj_memcpy(umem->umm_pool, media_addr, addr, n,
+			       PMEMOBJ_F_MEM_NODRAIN);
```
Comment (johannlombardi) [inline src/bio/bio_buffer.c:663]: from my local testing, pmem_memcpy_persist() provides a higher throughput than pmem_memcpy_nodrain(). We should probably use that instead.

Comment (NiuYawei) [reply src/bio/bio_buffer.c:663]: Yes, perhaps we'd change it back to persistent copy. I think that our tx commit will drain memory controller anyway, so there could be no much difference. I didn't change it in this patch, you could try change it back to persistent copy to see if it helps.

**Comment by mchaarawi** [comment]
@NiuYawei 
could you please rebase this patch with latest master?
thanks!

**Comment by NiuYawei** [comment]
> @NiuYawei
> could you please rebase this patch with latest master?
> thanks!

Rebased.

**Pull request closed without merging**